### PR TITLE
post: gradle-protobuf

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

cc @Xuanwo @ice1000

### CODE REVIEW

**Review by ice1000 — COMMENTED** [review]
牛逼

File: source/_posts/gradle-protobuf.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+```groovy
+plugins {
+    id 'java'
+    id 'com.google.protobuf' version '0.8.18'
+    id 'com.github.johnrengelman.shadow' version '7.1.2'
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply source/_posts/gradle-protobuf.md:66]:
```suggestion
```

[the remark this answers is not in the document]
Comment (tisonkun) [reply source/_posts/gradle-protobuf.md]: @ice1000 updated.

File: source/_posts/gradle-protobuf.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+date: 2022-03-08
+author: tison
+categories: tison
+tags:
+    - Gradle
+    - Java
+    - Protobuf
+---
+
+近日尝试利用 [Apache Ratis](https://github.com/apache/ratis) 这个项目包装一个 Raft 协议驱动的状态机的时候，遇到了需要用 Protobuf 传输数据的场景。由于 Gradle 构建工具的门槛和 Java 语言项目的某些惯例碰到了使用上的问题，这里记录一下我在这个玩具项目当中的用例。
```
Comment (Xuanwo) [inline source/_posts/gradle-protobuf.md:12]: "近日" 与 “的时候” 疑似语病

Comment (tisonkun) [reply source/_posts/gradle-protobuf.md:12]:
```suggestion
近日尝试利用 [Apache Ratis](https://github.com/apache/ratis) 这个项目包装一个 Raft 协议驱动的状态机，遇到了需要用 Protobuf 传输数据的场景。由于 Gradle 构建工具的门槛和 Java 语言项目的某些惯例碰到了使用上的问题，这里记录一下我在这个玩具项目当中的用例。
```

Comment (tisonkun) [reply source/_posts/gradle-protobuf.md]: Updated. Thanks!

**Review by Xuanwo — APPROVED** [review]

**Comment by ice1000** [comment]
tql

**Pull request merged**